python/cuml/manifold/umap.pyx

```diff
@@ -197,6 +200,13 @@ class UMAP(Base):
                 feature is made optional in the GPU version due to the
                 significant overhead in copying memory to the host for
                 computing the hash. (default = False)
+    random_state : int, RandomState instance or None, optional (default=None)
+        random_state is the seed used by the random number generator during
+        embedding initialization and during sampling used by the optimizer.
+        CAUTION : When random_state == None, the algorithm will run
+        a fully optimised version that produces inconsistent
+        results across runs. To produce consistent results,
+        please specify a random_state value.
```

Comment:
I like this extended description, though I'm not sure we need the `CAUTION` here. I think we might be well served to include the general trade-off of setting an explicit random state and not setting one. Something like 

```
Note: Unfortunately, achieving a high amount of parallelism during the optimization stage often comes at the expense of determinism, since many floating-point additions are being made in parallel without a deterministic ordering. This causes slightly different results across training sessions, even when the same seed is used for random number generation. Setting a random_state will enable consistency of trained embeddings, allowing for reproducible results to 3 digits of precision, but will do so at the expense of potentially slower training and increased memory usage.
```

Reply:
Ok